static/app/utils/replays/hooks/useIsLive.tsx

```diff
@@ -0,0 +1,39 @@
+import {useEffect, useState} from 'react';
+
+import type ReplayReader from 'sentry/utils/replays/replayReader';
+import useTimeout from 'sentry/utils/useTimeout';
+
+type Props = {
+  replayReader: ReplayReader | null;
+};
+
+export default function useIsLive({replayReader}: Props) {
+  const [isLive, setIsLive] = useState<boolean>(replayReader?.getIsLive() ?? false);
+
+  let TIME_UNTIL_NOT_LIVE_MS = 0;
+
+  if (replayReader) {
+    const TIME_UNTIL_REPLAY_FINISHED_MS =
+      replayReader.getStartTimestampMs() + 60 * 60 * 1000 - Date.now();
```

Comment:
minimal savings but i would precalc the hours in ms, and  you can use `_` as a separator for readability.

```suggestion
      replayReader.getStartTimestampMs() + 3_600_000 - Date.now();
```